File: D8_codes/Mixing_C_Fortran_codes/axpy.f90
! Created by G.P. Brandino for the Course P1.1 Scientific Programming Environment @ MHPC
! Last Revision: July 2015
!
! to be used in conjunction with call_axpy.c
! gfortran -c axpy.f90
! gcc call_axpy.c axpy.o

subroutine Axpy(n,a, arr1, arr2,arr3)
  integer,intent(in) :: n,a 
  integer, dimension(n), intent(in) :: arr1, arr2
  integer, dimension(n), intent(out) :: arr3

  arr3=a*arr1+arr2
  
end subroutine Axpy

File: D8_codes/Mixing_C_Fortran_codes/axpy.c
/* Created by G.P. Brandino for the Course P1.1 Scientific Programming Environment @ MHPC
 * Last Revision: July 2015
 *
 * to be used in conjunction with call_axpy.f90
 * gcc -c axpy.c
 * gfortran call_axpy.f90 axpy.o
 */


void axpy(int n, int a, int* arr1, int* arr2, int* arr3)
{
  int i;
  for (i=0;i<n;i++)
      {
      arr3[i]=a*arr1[i]+arr2[i];
      }
}
